tools/utils.py

```diff
@@ -263,3 +266,95 @@ def normalize_image(
     b = target_max - a * image.max()
     image_norm = (a * image + b).astype(target_type)
     return image_norm
+
+
+class BoundaryDataset(Dataset):
+    def __init__(self,
+                 img_paths: List[str],
+                 input_size: Union[int, List[int]] = (1024, 1024)):
+        self.img_paths = img_paths
+        self.input_size = (input_size, input_size) if isinstance(input_size, int) else input_size
+
+    def __len__(self):
+        return len(self.img_paths)
+
+    def __getitem__(self, idx):
+        img_path = self.img_paths[idx]
+        img = cv2.imread(img_path)
+        img = cv2.resize(img, self.input_size)
+        return img, img_path
+
+
+class MorphologicalTransformations:
+    def __init__(
+            self,
+            image_file: str,
+    ):
+
+        self.image_file = image_file
+        self.MAX_PIXEL = 255
+        self.MIN_PIXEL = 0
+
+    def read_image(self):
+        image_src = cv2.imread(self.image_file, 0)
+        return image_src
+
+    def binary(
+            self,
+            thresh_method: str = 'otsu',
+            thresh_val: float = None,
+    ):
+        color_1 = self.MAX_PIXEL
+        color_2 = self.MIN_PIXEL
+        image_src = self.read_image()
+
+        assert thresh_method in ['otsu', 'triangle', 'manual'], f'Invalid thresh_method: {thresh_method}'
+
+        if thresh_method == 'otsu':
+            threshold_value, _ = cv2.threshold(image_src, 0, 255, cv2.THRESH_BINARY + cv2.THRESH_OTSU)
```

Comment:
@sriva-e 

Does **_** (underscore variable) output binary image? If so, can we use it?

Reply:
![image](https://user-images.githubusercontent.com/89933560/184637532-fc7ade53-52c8-45e1-b5a8-9f23e5957e43.png)

This is what " _" outputs , it's a binary image